corehq/apps/domain/decorators.py

```diff
@@ -416,6 +425,20 @@ def api_auth(*, allow_creds_in_data=True, oauth_scopes=None):
         default=DIGEST,
         oauth_scopes=oauth_scopes,
         allow_creds_in_data=allow_creds_in_data,
+        allow_api_key_as_password=False,  # if supporting digest auth, you cannot use an api key as a password
+    )
+
+
+def api_auth_no_digest(*, allow_creds_in_data=True, oauth_scopes=None, allow_api_key_as_password=False):
+    """
+    Intentionally does not support digest auth to allow using api key as password if desired.
+    See determine_authtype_from_header for more details on how defaulting to BASIC drops DIGEST support
+    """
+    return get_multi_auth_decorator(
+        default=BASIC,
```

Comment:
I'm curious if defaulting to BASIC has any other side effects that may not be anticipated? I see we do this in other places such as mobile and odata auth decorators as well, so it seems safe.

Reply:
This seems pretty safe to me, as its locked behind this specific new decorator, which, to my knowledge, was never intended for digest authentication.